OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/fieldassignability/L0FieldAssignabilityAnalysis.scala

```diff
@@ -70,44 +93,82 @@ class L0FieldAssignabilityAnalysis private[analyses] (val project: SomeProject)
         if (field.classFile.methods.exists(_.isNative))
             return Result(field, Assignable);
 
-        val classFile = field.classFile
-        val thisType = classFile.thisType
-
-        for {
-            (method, pcs) <- fieldAccessInformation.writeAccesses(field)
-            if !method.isStaticInitializer
-            pc <- pcs
-        } {
-            method.body.get.instructions(pc) match {
-                case PUTSTATIC(`thisType`, fieldName, fieldType) =>
-                    // We don't need to lookup the field in the class
-                    // hierarchy since we are only concerned about private
-                    // fields so far... so we don't have to do a full
-                    // resolution of the field reference.
-                    val field = classFile.findField(fieldName, fieldType)
-                    if (field.isDefined) {
-                        return Result(field.get, Assignable);
-                    }
-
-                case _ =>
+        val faiEP = propertyStore(declaredFields(field), FieldWriteAccessInformation.key)
+        if (handleFieldWriteAccessInformation(faiEP))
+            return Result(field, Assignable)
+
+        createResult()
+    }
+
+    /**
+     * Processes the given field access information to evaluate if the given field is written statically in the method at
+     * the given PCs. Updates the state to account for the new value.
+     */
+    def handleFieldWriteAccessInformation(
+        faiEP: EOptionP[DeclaredField, FieldWriteAccessInformation]
+    )(implicit state: AnalysisState): Boolean = {
+        val assignable = if (faiEP.hasUBP) {
+            val (seenDirectAccesses, seenIndirectAccesses) = state.latestFieldWriteAccessInformation match {
+                case Some(UBP(fai)) => (fai.numDirectAccesses, fai.numIndirectAccesses)
+                case _              => (0, 0)
             }
+
+            faiEP.ub.getNewestAccesses(
+                faiEP.ub.numDirectAccesses - seenDirectAccesses,
+                faiEP.ub.numIndirectAccesses - seenIndirectAccesses
+            ) exists { wa =>
+                    val method = contextProvider.contextFromId(wa._1).method.definedMethod
+                    if (method.isStaticInitializer) {
+                        if (wa._3.isDefined) {
+                            // If a receiver is defined, we know that the access was not static
+                            // IMPROVE: Add static information to accesses and resolve this
+                            false
+                        } else {
+                            // As a fallback, we look for a trivial static write access in the statements
+                            method.body.get.instructions(wa._2).opcode == PUTSTATIC.opcode
```

Comment:
In theory, we could look if we encounter a trivial `PUTSTATIC`, otherwise assume its assignable...

But that is just the same as returning true here anyways. Will do.